costmodel/costmodel.go

```diff
@@ -1948,51 +1985,58 @@ func (cm *CostModel) costDataRange(cli prometheusClient.Client, clientset kubern
 	containers := make(map[string]bool)
 	otherClusterPVRecorded := make(map[string]bool)
 
-	RAMReqMap, err := GetContainerMetricVectors(resultRAMRequests, true, normalizationValue, clusterID)
+	RAMReqMap, err := GetNormalizedContainerMetricVectors(resultRAMRequests, normalizationValue, clusterID)
 	if err != nil {
 		return nil, err
 	}
 	for key := range RAMReqMap {
 		containers[key] = true
 	}
-	RAMUsedMap, err := GetContainerMetricVectors(resultRAMUsage, true, normalizationValue, clusterID)
+
+	RAMUsedMap, err := GetNormalizedContainerMetricVectors(resultRAMUsage, normalizationValue, clusterID)
 	if err != nil {
 		return nil, err
 	}
 	for key := range RAMUsedMap {
 		containers[key] = true
 	}
 
-	CPUReqMap, err := GetContainerMetricVectors(resultCPURequests, true, normalizationValue, clusterID)
+	CPUReqMap, err := GetNormalizedContainerMetricVectors(resultCPURequests, normalizationValue, clusterID)
 	if err != nil {
 		return nil, err
 	}
 	for key := range CPUReqMap {
 		containers[key] = true
 	}
-	CPUUsedMap, err := GetContainerMetricVectors(resultCPUUsage, false, normalizationValue, clusterID) // No need to normalize here, as this comes from a counter
+
+	// No need to normalize here, as this comes from a counter
+	// CPUUsedMap, err := GetContainerMetricVectors(resultCPUUsage, clusterID)
+	// TODO Is that ^ true? We were normalizing before (in spite of the comment) so I'm keeping it.
+	CPUUsedMap, err := GetNormalizedContainerMetricVectors(resultCPUUsage, normalizationValue, clusterID)
```

Comment:
So, before, we were passing the `normalizationValue`, but the second parameter to `GetContainerMetricVectors` is `normalize bool`, which we were passing `false`. The old line was:

```
                                          This is tells the method not to normalize
                                                              |
                                                              V
CPUUsedMap, err := GetContainerMetricVectors(resultCPUUsage, false, normalizationValue, clusterID)
```

The actual logic in the method:
```
               if normalize && normalizationValue != 0 {
			for _, v := range val.Values {
				v.Value = v.Value / normalizationValue
			}
		}
```

Reply:
So, I realized I made a mistake. The `GetContainerMetricVector` for non-ranged queries actually honors the `normalize bool` parameter while the `GetContainerMetricVectors` method didn't. I'm not sure if this was intended or not, and may have actually been a bug I introduced when moving to the `PromQueryResult` refactor. Going to investigate a bit more here.

Reply:
Tracked down my refactor changeset, and it doesn't look like we ever honored the `normalize bool` parameter in `GetContainerMetrics()` Here's the version from early November:

https://github.com/kubecost/cost-model/blob/95d335b75ebbc283efcd4d152c73bebbc3c1a5a1/costmodel/costmodel.go#L2563-L2615

However, if you go back before the existence of `vector.go`, there is a version where the logic was similar to `GetContainerMetricVector`:

https://github.com/kubecost/cost-model/blob/f7eb99baa6872f071081334059ac82477116c2d4/costmodel/costmodel.go#L1704-L1758